### PR TITLE
refactor: move presets to modal

### PR DESCRIPTION
**Pull request by haitmo**
image

_temporary look and feel of modal_

First Priority:
- [x] CRUD page
  - [x] Title
  - [x] Description
  - [x] Access
  - [x] Alert if one language is missing (nice to have)
- [x] Presets list
- [x] Add tests

For Later:
- [x] Favourites
- [ ] Tags
- [ ] Filters

Things to consider:

- Deleting a preset could affect defaults for shared presets?
- A chat sessions loads a copy of preset options _NOT_ the preset options themselves.

### CODE REVIEW

File: django/requirements_dev.txt [hunk 1]
```diff
@@ -7,3 +7,4 @@ django-debug-toolbar==4.3.0
 djlint==1.34.1
 watchdog==4.0.1
 rich==13.7.1
+django-livereload-server==0.5.1
```
Comment (haitmo) [inline django/requirements_dev.txt:10]: Temporarily testing this live reload library. Might not make it in.

File: django/chat/templates/chat/components/chat_options_sidebar.html [hunk 2]
```diff
@@ -7,7 +7,8 @@
   <div class="border-start h-100 position-relative">
     <button id="right-sidebar-resize-handle" class="btn btn-none" type="button">
       <i class="bi bi-grip-vertical"></i>
-      <span class="visually-hidden">{% trans "Drag or use the left and right arrows to resize." %}</span>
+      <span class="visually-hidden">{% trans "Drag or use the left and right arrows to
+      resize." %}</span>
```
Comment (juleskuehn) [inline django/chat/templates/chat/components/chat_options_sidebar.html:11]: Did something go wrong with DjLint? If these lines are split the translation tag will not work.

**Review by juleskuehn — COMMENTED** [review]
Looks great overall! I've resolved some merge conflicts and pushed to this branch.

Tested and working well:
- Save current settings as a new preset.
- Set a default preset.
- Opening a new chat uses the default preset you selected.
- Reset to default button
- Loading preset
- "Heart" (favorite) preset changes sorting order

The fade transition inside the modal is nice!

Here are some changes I'd like to see before merging.

1. Sidebar button location/labels:
image

This is the code copy/paste from the browser inspector for reference.
```html
<div class="row px-2 py-2">
  <div class="col pe-0" data-bs-toggle="modal" data-bs-target="#presets-modal">
    <button type="button" id="presets-modal-button" class="btn btn-sm btn-outline-dark w-100" data-bs-toggle="tooltip" data-bs-title="Open Presets Menu" hx-get="/chat/id/bb15a888-2fa1-446d-bfd1-9d9ba971a1a1/options/presets/" hx-target="#presets-modal-body" hx-swap="innerHTML" onclick="document.getElementById('presets-modal-body').innerHTML = '';">
      <span class="visually-hidden">Open Presets Menu</span>
      <i class="bi bi-folder"></i><span>Load</span>
    </button>
  </div><div class="col ps-1 pe-0" data-bs-toggle="modal" data-bs-target="#presets-modal">
    <button type="button" class="btn btn-sm btn-outline-dark w-100" data-bs-toggle="tooltip" data-bs-title="Create preset from current settings" hx-get="/chat/id/bb15a888-2fa1-446d-bfd1-9d9ba971a1a1/options/presets/form/" hx-target="#presets-modal-body" hx-swap="innerHTML transition:true">
      <i class="bi bi-floppy2-fill"></i>
      <span class="">Save</span>
    </button>
  </div>
  <div class="col ps-1">
    <button type="button" class="btn btn-sm btn-outline-dark w-100" data-bs-toggle="tooltip" data-bs-title="Reset settings to default" hx-get="/chat/id/bb15a888-2fa1-446d-bfd1-9d9ba971a1a1/options/preset/reset" hx-target="#options-accordion" hx-swap="outerHTML">
      <i class="bi bi-arrow-counterclockwise"></i><span>Reset</span>
    </button>
  </div>
</div>
```

2. Remove this tooltip
image

3. Consistent padding / capitalization
image
  - The margin under the second label is wrong. You need to use class "form-label".
  - Languages should be capitalized ("English", "French").
  - "Accessible to" should be in sentence case
  - "Available Presets" header should also be in sentence case. Actually, I don't think it serves much purpose so you could just remove this header.
Tooltips also should have consistent capitalization, e.g. this one is all Title Case but others are Sentence case:
image
  
4. Aesthetics of tabs - make relationship between tabs and content more clear
image
These are some of the changes I made in the screenshot above:
```html
    <ul class="nav nav-tabs nav-fill mx-3" id="preset-edit-tabs" role="tablist">
...
   
    <div class="tab-content rounded px-3 mb-3 border">
      ...
              <label class="form-label" for="id_description_en">Description (English)</label>
              ...
              (remove bottom margin on "accessible to" container)
  <div class="modal-footer pb-0 pt-1 border-0">
```

and css:
```
.modal-body {
  max-height: 85vh !important;
  overflow-y: auto;
}
```

5. Tooltips are acting weird.
image
Tooltips aren't going away after clicking a button. The JS console is showing some errors which may be relevant:
image
Also more generally would be nice for tooltips to disappear after clicked because sometimes weird things can happen otherwise (although this particular problem shouldn't happen after moving the buttons to the top)
image

6. Make it possible for Otto Admins to share presets publicly.
  - I can't see any way to do this other than going into Django shell and editing the DB object. As a result I also can't test this functionality.
  
7. Change "Presets" modal header to "Settings presets" so it is more clear.

8. Run load_app_localization and commit the changes.
  - Then test in French and make sure everything has a translation.
  
9. "Create new preset from current chat settings" button in the "browse presets" page (as shown in Hamza's mockup in this PR description)

10. Ability to update an existing preset with the current chat's settings. This could be a button next to "Delete" inside the "Edit" screen for an existing preset
image

Minor qualms:
- Tiles moving around after I click the star button (its ok if the default one goes to the top AFTER you next load the modal - but not right after you click the button. Its confusing.)
  - I think you can work around this by using JS to deselect the star in the other tiles rather than swapping the whole cards area (which, of course, will reorder them)
  - "Heart" button behaves as I would hope "default" does. It changes the sorting the next time you open the modal.
  
I will do a more thorough reading of the code in the next round, but I did skim and first impression is "LGTM". :)

**Review by haitmo — COMMENTED** [review]
Thanks Michel! A few comments:

- [ ] A few JavaScript errors seem to pop up, could you look into it?

image

- [ ] The tabs are no longer aligned with the text input

image

- [ ] Some labels and titles still appear in English on French pages

- [ ] A few of Jules's comments are still left unaddressed I believe. Here are a few I noticed
  - [x] A way for admins to make presets public
  - [ ] Styling of tabs (unless its just on my end?)
  - [ ] Adding the create preset button in the modal

**Review by mcusteau — APPROVED** [review]

**Pull request merged**